tests/modules/images/fetchpriority/Fetchpriority-Tests.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+class Fetchpriority_Tests extends WP_UnitTestCase {
+
+	public function test_wp_filter_content_tags_filter_adds_fetchpriority_fist_image_only() {
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		$img           = get_image_tag( $attachment_id, '', '', '', 'large' );
+
+		$this->assertStringContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add( $img, 'the_content' ) );
+		$this->assertStringNotContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add( $img, 'not_content' ) );
+
+		$img = str_replace( '<img ', '<img loading="lazy" ', $img );
+		$this->assertStringNotContainsString( 'fetchpriority="high"', fetchpriority_img_tag_add( $img, 'the_content' ) );
+	}
```

Comment:
See above regarding the function name.

Note also that the test name here is misleading, as this function does not test `wp_filter_content_tags()` at all. It only tests the new function added here, not how it affects `wp_filter_content_tags()`. It also doesn't test that it only adds the attribute to the first image only, so that also needs to be changed in the test name.

Also see my comment above on that we need to add a space before the attribute name when checking for it.

```suggestion
	public function test_fetchpriority_img_tag_add_attr_based_on_context_and_loading_lazy() {
		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
		$img           = get_image_tag( $attachment_id, '', '', '', 'large' );

		$this->assertStringContainsString( ' fetchpriority="high"', fetchpriority_img_tag_add_attr( $img, 'the_content' ) );
		$this->assertStringNotContainsString( ' fetchpriority="high"', fetchpriority_img_tag_add_attr( $img, 'not_content' ) );

		$img = str_replace( '<img ', '<img loading="lazy" ', $img );
		$this->assertStringNotContainsString( ' fetchpriority="high"', fetchpriority_img_tag_add_attr( $img, 'the_content' ) );
	}
```

Reply:
this is already updated in the code base